### PR TITLE
added support for statistics

### PR DESCRIPTION
**Pull request by edewit**
To update the statistics in OnLaunched e.Arguments is set when the notification was touched, then you can call TouchToOpen to update the statistics:

```cs
protected override void OnLaunched(LaunchActivatedEventArgs e) 
{
...
   if (!string.IsNullOrEmpty(e.Arguments))
   {
      await registration.TouchToOpen(pushConfig);
   }
```
be sure to add page param to the message as this is mandatory for windows to be able to use extra parameters.
```json
   "windows":{  
      "page":"/MainPage.xaml"
   }
```

### CODE REVIEW

File: aerogear-windows-push/IUPSHttpClient.cs [hunk 1]
```diff
@@ -26,5 +26,6 @@ namespace AeroGear.Push
     public interface IUPSHttpClient
     {
         Task<HttpStatusCode> register(Installation installation);
+        Task<HttpStatusCode> register(Installation installation, string pushMetricsId);
```
Comment (matzew) [inline aerogear-windows-push/IUPSHttpClient.cs]: mind do add some API doc ?

**Comment by matzew** [comment]
:+1: 

One thing, before we merge this (and other clients), how about we keep it on a branch until we all agree on the server side API hook (to avoid rounds and updates on the clients, in case something changes there)

**Comment by sebastienblanc** [comment]
@edewit Analytics is now on https://github.com/aerogear/aerogear-unifiedpush-server/pull/540 , for you the most important change is the rename of the header `aerogear-push-id`

File: aerogear-windows-push/Registration.cs [hunk 2]
```diff
@@ -25,28 +25,47 @@ namespace AeroGear.Push
     /// </summary>
     public abstract class Registration
     {
+        private const string CHANNEL_KEY = "Channel";
+        protected const string PUSH_ID_KEY = "push-identifier";
```
Comment (sebastienblanc) [inline aerogear-windows-push/Registration.cs]: Is now `aerogear-push-id`

File: aerogear-windows-push/UPSHttpClient.cs [hunk 3]
```diff
@@ -45,13 +45,28 @@ public UPSHttpClient(Uri uri, string username, string password)
...
         {
+            return await Request(installation, delegate(HttpWebRequest request) { });
+        }
+
+        public async Task<HttpStatusCode> register(Installation installation, string pushMetricsId)
+        {
+
+            return await Request(installation, delegate(HttpWebRequest request)
+            {
+                request.Headers["push-identifier"] = pushMetricsId;
```
Comment (matzew) [inline aerogear-windows-push/UPSHttpClient.cs]: See https://github.com/aerogear/aerogear-unifiedpush-server/pull/543 - we now have a different API for that.

Metrics is done by PUT agains a separate endpoint

**Comment by matzew** [comment]
done for AGPUSH-1235

**Comment by matzew** [comment]
thanks, @edewit ! landed on master (3861fee)

**Comment by matzew** [comment]
@edewit can we have a -beta.1 release (binary on nugget?) for this?

BTW. is there a JIRA ticket for updating the helloworld, showing the analtyics ?

**Comment by edewit** [comment]
[Update HelloWorld with metrics](https://issues.jboss.org/browse/AGWIN-5)
[nuget beta release](https://www.nuget.org/packages/aerogear-windows-push/0.0.9-beta1)

**Pull request merged**